addon/bootstrap.js

```diff
@@ -49,6 +50,21 @@ const prefObserver = {
   }
 };
 
+let historyEnabled = prefs.getBoolPref(HISTORY_ENABLED_PREF);
+// TODO: since checking "never remember history" requires a restart,
+// is it necessary to observe the places.history.enabled pref?
```

Comment:
I think it might be good to observe the pref, as it's not a lot of code, and the pref _could_ be changed in about:config without triggering a restart. If nothing else, this might make QA's life a little easier.

Reply:
We could also fetch it at the same time we fetch the telemetry pref, which we do before each major interaction. We could simply re-read it each time.